pkg/venafi/tpp/connector.go

```diff
@@ -678,19 +678,51 @@ func (c *Connector) proccessLocation(req *certificate.Request) error {
 	return nil
 }
 
-// RequestCertificate submits the CSR to TPP returning the DN of the requested Certificate
+// RequestCertificate requests the certificate to TPP and returns the DN of the
+// requested Certificate (this is called the "pickup ID" in the vcert CLI). If
+// the certificate is in the middle of an enrollment, the enrollment will be
+// reset before requesting (this doesn't affect the existing certificate if one
+// was already issued).
 func (c *Connector) RequestCertificate(req *certificate.Request) (requestID string, err error) {
 	if req.Location != nil {
 		err = c.proccessLocation(req)
 		if err != nil {
 			return
 		}
 	}
+
+	// Let's unconditionally reset any prior pending or failed enrollment. This
+	// doesn't affect the existing certificate if one was already issued. At
+	// first, we wanted to decide whether to reset or not by first retrieving
```

Comment:
Hi , I am not sure if I understand the rational behind reseting it every time.  This is edge case we are trying to solve here. 
I would expect Reset to only happen if certificate request is failing with certain error. This seems like an unnecessary API call every time when majority of time it wont be needed. 

Request Cert  --> If Failed with Error --> Reset --> Request again one more time. 
Request cert --> if Good --> Return

Reply:
I realize I haven't documented the reason why I am not first requesting, and then resetting if that fails.

The reason it is not possible is that `POST  /request` always succeeds as long as the given CSR or certificate parameters are valid. Calling `POST /request` doesn't allow you to know whether `POST /reset` needs to be called or not.

Does that make sense?

Reply:
Since we settled on "Solution 3", I will resolve this conversation.